src/useNotification.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import * as thisModule from './useNotification';
+
+export async function requestPermission(setPermission) {
+  const permissionFromUser = await Notification.requestPermission();
+
+  setPermission(permissionFromUser);
+}
```

Comment:
It could be that after the `await` the component is already unmounted. Something like this should work:

```ts
export async function requestPermission(setPermission, isMounted) {
  const permissionFromUser = await Notification.requestPermission();
  if (!isMounted()) return;
  setPermission(permissionFromUser);
}
```